File: LeetCode/Study Plan/Practice Problems/Easy/Shuffle The Array/Go/shuffleTheArray.go
package main

import "fmt"

func shuffle(nums []int, n int) []int {
	size := 2 * n
	i, j := 0, n

	res := make([]int, size)
	k := 0
	for i < n && j < size {
		res[k] = nums[i]
		res[k+1] = nums[j]
		k += 2
		i++
		j++
	}

	return res
}

func main() {
	fmt.Println(shuffle([]int{2, 5, 1, 3, 4, 7}, 3))
}

File: LeetCode/Easy/Shuffle The Array/Java/shuffleTheArray.java
class Solution {
    public int[] shuffle(int[] nums, int n) {
        int size = 2*n;
        int i=0, j=n;

        int[] res = new int[size];
        int k = 0;
        while(i < n && j < size) {
            res[k] = nums[i++];
            res[k+1] = nums[j++];
            k+=2;
        }
        return res;
    }
}
